st2api/st2api/app.py

```diff
@@ -52,6 +54,16 @@ def setup_app(config=None):
 
     is_gunicorn = getattr(config, 'is_gunicorn', False)
     if is_gunicorn:
+        # Note: We need to perform monkey patching in the worker. If we do it in
+        # the master process (gunicorn_config.py), it breaks tons of things
+        # including shutdown
+        eventlet.monkey_patch(
```

Comment:
To clarify - monkey patching is really only needed if `sync` gunicorn worker is used. If async one is used, the worker class itself already performs monkey patching.

In any case, I think it's probably still a good idea to leave it here so we can easily switch between sync / async workers without making any code changes (double monkey patch is a no-op).